apps/admin-gui/src/app/users/pages/user-detail-page/user-overview/user-overview.component.ts

```diff
@@ -13,14 +21,41 @@ export class UserOverviewComponent implements OnInit {
 
   @HostBinding('class.router-component') true;
 
-  constructor() { }
+  constructor(private userService: UsersManagerService,
+              private attributeService: AttributesManagerService,
+              private storeService: StoreService,
+              private route: ActivatedRoute,
+              private dialog: MatDialog) { }
 
   navItems: MenuItem[] = [];
   user: User;
+  userID: number;
   path: string;
+  mailDataSource: MatTableDataSource<Attribute>;
+  displayedColumns = ['name', 'value'];
+  inMyProfile = false;
+  preferredMail: Attribute;
 
   ngOnInit() {
-    this.initNavItems();
+    this.route.params.subscribe(params => {
+      if(params['userId'] !== undefined) {
+        this.userService.getUserById(params['userId']).subscribe(user => {
+          this.user = user;
+
+          this.initNavItems();
+        });
+      } else {
+        this.inMyProfile = true;
+        this.userID = this.storeService.getPerunPrincipal().user.id;
+
+        this.attributeService.getUserAttributeByName(this.userID, Urns.USER_DEF_PREFERRED_MAIL).subscribe(mail => {
+          this.preferredMail = mail;
+
+          this.mailDataSource = new MatTableDataSource<Attribute>([this.preferredMail]);
```

Comment:
what if the prefferedmail is null? would the gui act atleast somehow normal? or it would broke everything?
its a rare situation but it could happen